conda_lock/pypi_solver.py

```diff
@@ -55,6 +59,12 @@ class PlatformEnv(Env):
     Fake poetry Env to match PyPI distributions to the target conda environment
     """
 
+    _sys_platform: Literal["darwin", "linux", "win32"]
```

Comment:
What is this for?

Reply:
This is a type annotation, so if you're running a type checker like `mypy` or `pyright` it will know what are the valid values of `_sys_platform`. I find it really useful because if you're in another file and you hover over a variable it'll show you the type and hence the possible values, making it much easier to figure out what's going on without needing to infer it from reading the code.

So in this case I was reading the code, I figured out that these are the possible values, and I declared them as such. And now the type checker statically verifies this and can offer hints.